br/pkg/lightning/backend/local/local.go

```diff
@@ -1541,19 +1664,54 @@ func (local *local) writeAndIngestByRanges(ctx context.Context, engine *Engine,
 	var allErr error
 	var wg sync.WaitGroup
 	metErr := atomic.NewBool(false)
+	rangeAndRegions := make([]RangeAndRegion, 0)
+	regionProcessingMap := sync.Map{}
 
 	for _, r := range ranges {
 		startKey := r.start
 		endKey := r.end
+		regions, err := local.scanRegion(ctx, engine, startKey, endKey, regionSplitSize, regionSplitKeys)
+		if err != nil {
+			log.FromContext(ctx).Warn("scaen regino failed", zap.Error(err))
+			return err
+		}
+		rr := RangeAndRegion{
+			r:           r,
+			isScheduled: false,
+			regions:     make(map[uint64]int),
+		}
+		for _, region := range regions {
+			for _, peer := range region.Region.Peers {
+				if val, ok := rr.regions[peer.StoreId]; ok {
```

Comment:
in golang we can make use of the zero value, directly `++` even for non-existing entries.

```go
package main

func main() {
	m := map[int]int{}
	m[2]++
	println(m[2])
}
```